spec/_attachments/ic.did

```diff
@@ -2,244 +2,305 @@ type canister_id = principal;
-      controllers : vec principal;
-  });
-  delete_canister : (record {canister_id : canister_id}) -> ();
-  deposit_cycles : (record {canister_id : canister_id}) -> ();
-  raw_rand : () -> (blob);
-  http_request : (record {
-    url : text;
-    max_response_bytes: opt nat64;
-    method : variant { get; head; post };
-    headers: vec http_header;
-    body : opt blob;
-    transform : opt record {
-      function : func (record {response : http_response; context : blob}) -> (http_response) query;
-      context : blob
-    };
-  }) -> (http_response);
-
-  // Threshold ECDSA signature
-  ecdsa_public_key : (record {
-    canister_id : opt canister_id;
-    derivation_path : vec blob;
-    key_id : record { curve: ecdsa_curve; name: text };
-  }) -> (record { public_key : blob; chain_code : blob; });
-  sign_with_ecdsa : (record {
-    message_hash : blob;
-    derivation_path : vec blob;
-    key_id : record { curve: ecdsa_curve; name: text };
-  }) -> (record { signature : blob });
-
-  // bitcoin interface
-  bitcoin_get_balance: (get_balance_request) -> (satoshi);
-  bitcoin_get_balance_query: (get_balance_request) -> (satoshi) query;
-  bitcoin_get_utxos: (get_utxos_request) -> (get_utxos_response);
-  bitcoin_get_utxos_query: (get_utxos_request) -> (get_utxos_response) query;
-  bitcoin_send_transaction: (send_transaction_request) -> ();
-  bitcoin_get_current_fee_percentiles: (get_current_fee_percentiles_request) -> (vec millisatoshi_per_byte);
-
-  // metrics interface
-  node_metrics_history : (record {
-    subnet_id : principal;
-    start_at_timestamp_nanos: nat64;
-  }) -> (vec record {
-    timestamp_nanos : nat64;
-    node_metrics : vec node_metrics;
-  });
-
-  // provisional interfaces for the pre-ledger world
-  provisional_create_canister_with_cycles : (record {
-    amount: opt nat;
-    settings : opt canister_settings;
-    specified_id: opt canister_id;
-    sender_canister_version : opt nat64;
-  }) -> (record {canister_id : canister_id});
-  provisional_top_up_canister :
-    (record { canister_id: canister_id; amount: nat }) -> ();
-}
+	create_canister : (create_canister_request) -> (create_canister_response);
+	update_settings : (update_settings_request) -> ();
+	upload_chunk : (upload_chunk_request) -> (chunk_hash);
+	clear_chunk_store : (clear_chunk_store_request) -> ();
+	stored_chunks : (stored_chunks_request) -> (vec chunk_hash);
+	install_code : (install_code_request) -> ();
+	install_chunked_code : (install_chunked_code_request) -> ();
+	uninstall_code : (uninstall_code_request) -> ();
+	start_canister : (start_canister_request) -> ();
+	stop_canister : (stop_canister_request) -> ();
+	canister_status : (canister_status_request) -> (canister_status_response);
+	canister_info : (canister_info_request) -> (canister_info_response);
+	delete_canister : (delete_canister_request) -> ();
+	deposit_cycles : (deposit_cycles_request) -> ();
+	raw_rand : () -> (blob);
+	http_request : (
+		record {
```

Comment:
should we have a type synonym for this and some other records that are still inlined in the service definition? my preference would be to  introduce type synonyms for all records in the service